src/Transport/Sendmail.php

```diff
@@ -260,6 +260,9 @@ protected function prepareParameters(Mail\Message $message)
         }
 
         $parameters = (string) $this->parameters;
+        if (preg_match('/^\-f.+$/', $parameters) || strpos($parameters, ' -f') !== false) {
```

Comment:
```suggestion
        if (strpos($parameters, '-f') !== false) {
```

This accomplishes the same thing, with one less operation.

Reply:
@weierophinney it doesn't seem right, imo

it's not the same, previously we accept `-f` at the beginning of when it is after a space.

I can imagine case `my-fancy-param` where `-f` is no at the beginning nor after a space.

Reply:
Then do it as follows:

```php
if (preg_match('/(^| )\-f.+/', $parameters)) {
```

It doesn't make sense to have two conditions when it's essentially the same condition ("is there a -f switch provided?"). The above regex does the "at the beginning" as well as "following a space" conditions.

Reply:
that's better ;-) there is also redundant `\` before `-` in the regexp

Reply:
@weierophinney updated.